File: app/src/main/java/org/michaelbel/moviemade/ui/base/PaddingItemDecoration.java
package org.michaelbel.moviemade.ui.base;

import android.graphics.Rect;
import android.view.View;

import androidx.recyclerview.widget.RecyclerView;

@SuppressWarnings("all")
public class PaddingItemDecoration extends RecyclerView.ItemDecoration {

    private int itemOffset;

    public PaddingItemDecoration() {}

    public PaddingItemDecoration(int itemOffset) {
        this.itemOffset = itemOffset;
    }

    public PaddingItemDecoration setOffset(int itemOffset) {
        this.itemOffset = itemOffset;
        return this;
    }

    @Override
    public void getItemOffsets(Rect outRect, View view, RecyclerView parent, RecyclerView.State state) {
        super.getItemOffsets(outRect, view, parent, state);
        outRect.set(itemOffset, itemOffset, itemOffset, itemOffset);
    }
}

File: app/src/main/java/org/michaelbel/moviemade/ui/base/PaddingItemDecoration.kt
package org.michaelbel.moviemade.ui.base

import android.graphics.Rect
import android.view.View

import androidx.recyclerview.widget.RecyclerView

class PaddingItemDecoration : RecyclerView.ItemDecoration {

    private var itemOffset: Int = 0

    constructor()

    constructor(itemOffset: Int) {
        this.itemOffset = itemOffset
    }

    fun setOffset(itemOffset: Int): PaddingItemDecoration {
        this.itemOffset = itemOffset
        return this
    }

    override fun getItemOffsets(outRect: Rect, view: View, parent: RecyclerView, state: RecyclerView.State) {
        super.getItemOffsets(outRect, view, parent, state)
        outRect.set(itemOffset, itemOffset, itemOffset, itemOffset)
    }
}